src/skan/napari_skan.py

```diff
@@ -1,29 +1,78 @@
 import numpy as np
 from enum import Enum
 from skimage.morphology import skeletonize
-
+from skan import summarize, Skeleton
+import napari
+from magicgui.widgets import Container, ComboBox, PushButton, Label, create_widget
+from napari.layers import Shapes 
 class SkeletonizeMethod(Enum):
-    Zhang = "zhang"
-    Lee = "lee"
-
-
-def skeletonize_labels(labels: "napari.types.LabelsData", method: SkeletonizeMethod) -> "napari.types.LabelsData":
-    """Takes a labels layer and a skimage skeletonize method and generates a skeleton representation
-
-    Parameters
-    ----------
-    labels : napari.types.LabelsData
-        A labels layer containing data to skeletonize
-    method : SkeletonizeMethod
-        Enum denoting the chosen skeletonize method method
-
-    Returns
-    -------
-    napari.types.LabelsData
-        Labels layer depecting the extracted skeleton
-    """
-    binary_labels = (labels > 0).astype(np.uint8)
-    # skeletonize returns a binary array, so we can just multiply it with the
-    # labels to get appropriate colors
-    skeletonized = skeletonize(binary_labels, method=method.value) * labels
-    return skeletonized
+    zhang = "zhang"
+    lee = "lee"
+
+def get_skeleton(labels: "napari.layers.Labels", choice: SkeletonizeMethod) -> "napari.layers.Shapes":
+    binary_labels = (labels.data > 0).astype(np.uint8)
+    binary_skeleton = skeletonize(binary_labels, method=choice.value)
+    
+    skeleton = Skeleton(binary_skeleton)
+
+    all_paths = [skeleton.path_coordinates(i)  
+            for i in range(skeleton.n_paths)]
+
+    return Shapes(all_paths,
+        shape_type='path',
+        edge_colormap='tab10',
+        metadata={"skeleton": skeleton}
+        )
```

Comment:
This should instead be a [LayerDataTuple](https://napari.org/dev/guides/magicgui.html?highlight=layerdatatuple#return-annotations) so we don't need to depend on napari.

```python
return (
        all_paths,
        {'shape_type': 'path', 'edge_colormap': 'tab10', 'metadata': {'skeleton': skeleton}},
	'shapes',
        )
```

I also think you should analyze the skeleton in this function and add the features to the layer — the next widget is way too complicated. This way we don't need to depend on magicgui either!

Reply:
@jni Updated to `LayerDataTuple` and now computing the summary at time of skeletonization. 
I think we still need some logic in case the user selects a shape layer without a skeleton feature, but I've tried to simplify.
The only way we can get rid of magicgui is if definitely knew the column names. We COULD hard code this but there is a parameter  `summarize` that returns an additional column.
I also removed the `get_features` button but now I'm having trouble populating the features combo on widget initialization. Maybe we can catch up this evening?